### ISSUE TITLE
Values seem to be out of date, any up to date fork?

### ISSUE THREAD

**Issue by NotRanged** [doxxx/ffxiv-craft-opt-web#419]
More of a question than an issue, but the values I get from the solver here seem to mismatch what the game gives even for static stuff. Is there an up to date fork of this project hosted somewhere?

**Comment by NotRanged** [comment]
I made my own.

https://github.com/NotRanged/NotRanged.github.io (code)
https://notranged.github.io./#/simulator (actual tool here)

A fork off of @wundaige 's version, I made changes to the genetic algorithm.

The main population now gets divided into subpopulations (you can set how many in options), that get culled after x generations of stagnation (also settable). It's best used by setting a large population (e.g. 12,000 divided into 6 subpops) and then you just leave it running for a few minutes before pressing stop.

I also added a new mutator to the genetic algorithm, reversal, after doing some research into good mutators to add (https://arxiv.org/ftp/arxiv/papers/1203/1203.3099.pdf). It seems to have improved algorithm performance a decent bit.

You'll find a few more extra settings in the solver options as well, and the recipe list has been updated to include 4-star level 80s and the likes.

**Comment by Raos123** [comment]
> 
> 
> I made my own.
> 
> https://github.com/NotRanged/NotRanged.github.io
> https://notranged.github.io./#/simulator
> 
> A fork off of @wundaige 's version, I made changes to the genetic algorithm.
> 
> The main population now gets divided into subpopulations (you can set how many in options), that get culled after x generations of stagnation (also settable). It's best used by setting a large population (e.g. 12,000 divided into 6 subpops) and then you just leave it running for a few minutes before pressing stop.
> 
> I also added a new mutator to the genetic algorithm, reversal, after doing some research into good mutators to add (https://arxiv.org/ftp/arxiv/papers/1203/1203.3099.pdf). It seems to have improved algorithm performance a decent bit.
> 
> You'll find a few more extra settings in the solver options as well, and the recipe list has been updated to include 4-star level 80s and the likes.

you are a god send! I was just trying to run my own local copy to do the same but for the love of god cant get the docker instructions to work without error.

**Comment by vitharr137** [comment]
This is quite nice, thank you! Would there be any way to introduce the custom recipe option from wunaidge's? That is my favorite addition to it, but these algorithm changes are incredible, too

**Comment by NotRanged** [comment]
Well, 3 months late but it should have custom recipes now. I haven't tested them yet but it's a direct copy from what wundaige made.

But yeah, once again:

The best way to use my version of the tool, is by leaving it running for a few minutes with only guaranteed actions, until you're satisfied or until you believe that it won't improve much further.

**Comment by NotRanged** [comment]
New update:

Added combo actions, some minor bugfixes.

The genetic algorithm really likes to rip up actions in sequence with every mutation, making combos like 'Observe -> Focused Synthesis' really rare to occur from this tool, despite being rather useful and powerful combos. I added in support by having them as singular actions, that then get simulated as their respective abilities and put into macro text properly as well.

**Comment by tlai34** [comment]
Do you have any of the collectables in the recipe list?

**Comment by NotRanged** [comment]
Yes, search for 'rarefied', it should have all of them. Do keep in mind that you can only craft up to 5 levels above what your current selected crafter's level is set to.

**Comment by vitharr137** [comment]
> 
> 
> New update:
> 
> Added combo actions, some minor bugfixes.
> 
> The genetic algorithm really likes to rip up actions in sequence with every mutation, making combos like 'Observe -> Focused Synthesis' really rare to occur from this tool, despite being rather useful and powerful combos. I added in support by having them as singular actions, that then get simulated as their respective abilities and put into macro text properly as well.

This works amazingly, my only recommendation would be a possible addition of the Basic/Standard Touch combo for the reduced cost. Thank you!

**Comment by Mike-Plouffe** [comment]
Thanks for updating so this works with newer stuff. Any chance of fixing the underlying issues though? Even the original version makes macros that are 2-3x as long as they should be
